### PR TITLE
Migrate to Java 8

### PR DESCRIPTION
**Pull request by nilols**
Start migrate the code to use Java 8 functionality

@KristianKarl

### CODE REVIEW

File: graphwalker-io/src/main/java/org/graphwalker/io/factory/dot/DotContextFactory.java [hunk 1]
```diff
@@ -190,11 +190,8 @@ public String getAsString(List<Context> contexts) {
   public void write(List<Context> contexts, Path path) throws IOException {
     File folder = path.toFile().getAbsoluteFile();
     Path dotFile = Paths.get(folder.toString(), contexts.get(0).getModel().getName() + ".dot");
-    OutputStream outputStream = Files.newOutputStream(dotFile);
-    try {
+    try (OutputStream outputStream = Files.newOutputStream(dotFile)) {
       outputStream.write(String.valueOf(getAsString(contexts)).getBytes());
-    } finally {
-      outputStream.close();
```
Comment (KristianKarl) [inline graphwalker-io/src/main/java/org/graphwalker/io/factory/dot/DotContextFactory.java]: These finally blocks was to satisfy Coverity static code analysis. Where an unclosed stream was a potential resource leak.
Has Java 8 changed this do you think?
Definitely not a biggie.

Comment (KristianKarl) [inline graphwalker-io/src/main/java/org/graphwalker/io/factory/dot/DotContextFactory.java]: I found the Coverity report:
```
*** CID 124364:  Resource leaks  (RESOURCE_LEAK)
/graphwalker-io/src/main/java/org/graphwalker/io/factory/dot/DotContextFactory.java: 180 in org.graphwalker.io.factory.dot.DotContextFactory.write(java.util.List, java.nio.file.Path)()
174             str.append("\"];").append(newLine);
175           }
176           str.append("}").append(newLine);
177
178           File folder = path.toFile().getAbsoluteFile();
179           Path dotFile = Paths.get(folder.toString(), context.getModel().getName() + ".dot");
>>>     CID 124364:  Resource leaks  (RESOURCE_LEAK)
>>>     Failing to save or close resource created by "java.nio.file.Files.newOutputStream(dotFile)" leaks it.
180           Files.newOutputStream(dotFile).write(String.valueOf(str).getBytes());
181         }
182       }
```

Comment (nilols) [inline graphwalker-io/src/main/java/org/graphwalker/io/factory/dot/DotContextFactory.java]: didn't know about that, will have a look and fix that

Comment (nilols) [inline graphwalker-io/src/main/java/org/graphwalker/io/factory/dot/DotContextFactory.java]: try-with-resource should handle closing of the stream, is that an old coverity report? or is it for this branch?

Comment (nilols) [inline graphwalker-io/src/main/java/org/graphwalker/io/factory/dot/DotContextFactory.java]: old I guess, last scan was Mar 05, 2017

Comment (nilols) [inline graphwalker-io/src/main/java/org/graphwalker/io/factory/dot/DotContextFactory.java]: I see in the coverity dashboard that there are two more issues, I can fix them and try to trigger a new check before we merge this

Comment (KristianKarl) [inline graphwalker-io/src/main/java/org/graphwalker/io/factory/dot/DotContextFactory.java]: :-)

**Comment by KristianKarl** [comment]
:+1:

**Comment by nilols** [comment]
The last coverity scan detected two issues, one was that getParent(), in the isolated class loader, could possible be called on a null object, this should now be solved by checking that the object is not null before calling the getParent() method. The other issue exist in the generated HelpMojo class, I checked the generated code and the issue should not be able to happen, either a later maven version have fixed this issue or the detection is wrong. I think we have to let the coverity scan execute again, to see if this is still a issue, before we can act upon it

**Pull request merged**